srtcore/srt.h

```diff
@@ -74,23 +74,44 @@ written by
 // for all lesser standard use compiler-specific attributes).
 #if defined(SRT_NO_DEPRECATED)
 
+// Unused: do not issue an unused variable warning
 #define SRT_ATR_UNUSED
+
+// Deprecated: function or symbol is deprecated
+// The *_PX version is the prefix attribute, which applies only
+// to functions (Microsoft compilers).
+
+// When deprecating a function, mark it:
+//
+// SRT_ATR_DEPRECATED_PX retval function(arguments) SRT_ATR_DEPRECATED;
+//
 #define SRT_ATR_DEPRECATED
+#define SRT_ATR_DEPRECATED_PX
+
+// Nodiscard: issue a warning if the return value was discarded.
```

Comment:
This is introduced because some compilers use prefix-type modifier, others use postfix-type modifier. This provides the attribute support for Microsoft compilers.